api/v1alpha1/ext_authz_type.go

```diff
@@ -0,0 +1,12 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// ExtAuthz defines the configuration for External Authorization.
+type ExtAuthz struct {
+	// GRPCURI defines the gRPC cluster name to use in that route.
+	GRPCURI string `json:"grpcURI,omitempty" yaml:"grpcURI"`
```

Comment:
It was easier to handle before my change in `url2Cluster()` since TLS was the only allowed scheme.
We need a way to pass the TLS state now, I agree URI usually just reflect host:port not the scheme, so the naming could be better or we need a boolean for TLS.

Reply:
@arkodg After taking a look at [the ExtAuthz configuration](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto#envoy-v3-api-msg-extensions-filters-http-ext-authz-v3-httpservice), I think we probably need a more sophisticated way to configure the  External Auth Service. The URL approach only allows users to configure the URL of the External Auth Service. It doesn't allow users to configure the other features that are different between the HTTP and GRPC service. The following is the configuration of the HTTP External Auth Service:

HTTP example:
```yaml
http_filters:
  - name: envoy.filters.http.ext_authz
    typed_config:
      "@type": type.googleapis.com/envoy.extensions.filters.http.ext_authz.v3.ExtAuthz
      http_service:
          server_uri: # Sets the HTTP server URI which the authorization requests must be sent to.
          path_prefix" # Sets a prefix to the value of authorization request header Path.
         authorization_request # Settings used for controlling authorization request metadata.
         authorization_response # Settings used for controlling authorization response metadata.
```

Reply:
do we need to support all these settings ?

Reply:
At least some of them, for example, I know a use case from one of our users that needs to specify the request headers to be added.